python/pacemaker/_cts/tests/cibsecret.py

```diff
@@ -0,0 +1,176 @@
+"""Test managing secrets with cibsecret."""
+
+__all__ = ["CibsecretTest"]
+__copyright__ = "Copyright 2025 the Pacemaker project contributors"
+__license__ = "GNU General Public License version 2 or later (GPLv2+) WITHOUT ANY WARRANTY"
+
+from pacemaker._cts.tests.ctstest import CTSTest
+from pacemaker._cts.tests.simulstartlite import SimulStartLite
+from pacemaker._cts.tests.starttest import StartTest
+from pacemaker._cts.timer import Timer
+
+# Disable various pylint warnings that occur in so many places throughout this
+# file it's easiest to just take care of them globally.  This does introduce the
+# possibility that we'll miss some other cause of the same warning, but we'll
+# just have to be careful.
+
+# pylint doesn't understand that self._env is subscriptable.
+# pylint: disable=unsubscriptable-object
+
+
+class CibsecretTest(CTSTest):
```

Comment:
Since this doesn't test `cibsecret sync`, this test only directly uses the local node. Do we benefit from a cts-lab test for this? It seems like we could create `cts-cli` tests where we set the `CIB_file` environment variable. We'd probably have to restrict it to run as root so that `cibsecret` can create the necessary directories/files -- so disable those tests by default.

The "set" test will try to sync the files, but we don't get any feedback on whether that succeeds, or on what the state of the secret on the other nodes is.

It is true that the sync fails (causing the set to fail) if there are no peers. I have pointed out in https://github.com/ClusterLabs/pacemaker/pull/3929/files#r2255597417 that I think that's a mistake.

Reply:
The `sync` subcommand doesn't seem strictly necessary - the various subcommands will still call down into `sync_one_file` when they do anything which propagates the secrets to all the other nodes.  `sync` seems handy if you've made changes manually, outside of cibsecret's management, and want them to be propagated as well.

I've added a test for that, as well as additional checks to test that the file contents are as expected on each node.

Reply:
> Yeah I agree. And I was just thinking about "why does this even need to be a cts-lab test?" since you had noted that "this one doesn't seem completely in the spirit of ctslab."

I forgot that the shell script requires Pacemaker to be running, which isn't suitable for cts-cli. Although, the C program *doesn't*. I made a note of that in another comment just now.